modulo5/projeto-labeBank/src/index.ts

```diff
@@ -0,0 +1,166 @@
+console.log("Hello, world!")
+
+import { ok } from 'assert';
+import express from 'express'
+import { Request, Response } from 'express'
+import { AddressInfo } from "net";
+import { restart } from 'nodemon';
+
+/* enum ACCTYPE {
+  corrente= "corrente",
+  poupança="poupança"
+} */
+
+
+
+type transaction={
+  tipo:string,
+  valor:number,
+  data:string 
+}
+
+type conta ={
+  userId:number,
+  name:string,
+  idade:number
+  cpf:string|number,
+  accType: string,
+  saldo:number,
+  extrato:transaction[]
+  
+}
+
+
+
+
+
+const contas :conta[]=[
+  {
+    userId:1,
+    name:"Lucas",
+    idade: 31,
+    cpf:"222.222.222.22",
+    accType: "corrente",
+    saldo: 1000,
+    extrato:[{
+      tipo: "debito",
+      valor: 45,
+      data: "25/02/1991"
+    },{
+      tipo:"debito",
+      valor:15,
+      data: "25/02/1991"
+
+    },{
+      tipo:"credito",
+      valor:200,
+      data: "25/02/1991"
+
+    }], 
+  },
+/* 
+  {
+    userId:1,
+    name:"Lucas",
+    idade: 31,
+    cpf:"222.222.222.21",
+    accType: "poupança",
+    extrato:[{
+      tipo: "debito",
+      valor: 45
+    }],
+ 
+  },
+
+  {
+    userId:2,
+    name:"Joao",
+    idade: 31,
+    cpf:"111.111.111-35",
+    accType: "poupança",
+    extrato:[{
+      tipo: "debito",
+      valor: 45
+    }],
+  },
+
+  {                               
+    userId:3,
+    name:"Carlos",
+    idade: 40,
+    cpf:"333.333.333-35",
+    accType: "corrente",
+    extrato:[{
+      tipo: "debito",
+      valor: 45
+    }],
+  },  */
+]
+
+
+
+const app = express();
+
+app.use(express.json());
+
+
+
+app.get("/ping", (req, res) => {          
+  res.send("Pong! 🏓")
+})
+
+app.get("/contas", (req: Request, res: Response) => {
+  res.status(201).send(contas)
+})  
+
+app.post("/contas", (req: Request, res: Response) => {
+  const data = (req.body)
```

Comment:
Normalmente, devemos passar cada item que será necessário no body. Pois assim conseguimos dar uma melhor informação ao usuário da api sobre o que é necessário para ser utilizado e também enviar mensagens de erro. 

O melhor seria criar as variáveis e armazenar o body delas. Uma variável para cada informação que for receber pelo body. 

const name  = req.body.name 
const idade =  req.body.idade
const cpf =  req. body.cpf 

 Por exemplo.